### PR TITLE
LP-1720: Added support of SSO to cater popup redirection

### PR DESCRIPTION
**Pull request by saad-waqar**
Added support of SSO to cater popup redirection

### CODE REVIEW

File: common/djangoapps/third_party_auth/pipeline.py [hunk 1]
```diff
@@ -560,7 +564,9 @@ def should_force_account_creation():
         elif auth_entry == AUTH_ENTRY_REGISTER:
             # User has authenticated with the third party provider and now wants to finish
             # creating their edX account.
-            return dispatch_to_register()
+            register_origin_url = strategy.request.session.get('next')
+            return dispatch_to_custom_redirect(register_origin_url) \
```
Comment (emraan004) [inline common/djangoapps/third_party_auth/pipeline.py]: Can we redirect from here instead of function?

Comment (saad-waqar) [reply common/djangoapps/third_party_auth/pipeline.py]: @emraan004 goes with the flow of how it is previously written. If you still think we should redirect directly from here, I can make that change

**Comment by saad-waqar** [comment]
When the popups are made, we have to add this in the context processor for all
`_third_party_auth_context(request, request.path)`

File: common/djangoapps/third_party_auth/pipeline.py (hunk 1 above)
Comment (emraan004) [reply common/djangoapps/third_party_auth/pipeline.py]: It felt unnecessary, but if this is how it was implemented, then it is fine for me. I'm resolving the conversation.

**Pull request merged**